internal/provider/provider.go

```diff
@@ -120,110 +120,84 @@ func (p *provider) Configure(ctx context.Context, req tfsdk.ConfigureProviderReq
 	p.client = c
 }
 
-func (p *provider) GetResources(_ context.Context) (map[string]tfsdk.ResourceType, diag.Diagnostics) {
-	return map[string]tfsdk.ResourceType{
-		"axual_user":                               userResourceType{},
-		"axual_group":                              groupResourceType{},
-		"axual_environment":                        environmentResourceType{},
-		"axual_topic":                              topicResourceType{},
-		"axual_topic_config":                       topicConfigResourceType{},
-		"axual_application":                        applicationResourceType{},
-		"axual_application_principal":              applicationPrincipalResourceType{},
-		"axual_application_deployment":             applicationDeploymentResourceType{},
-		"axual_application_access_grant":           applicationAccessGrantResourceType{},
-		"axual_application_access_grant_approval":  applicationAccessGrantApprovalResourceType{},
-		"axual_application_access_grant_rejection": applicationAccessGrantRejectionResourceType{},
-		"axual_schema_version":                     schemaVersionResourceType{},
-	}, nil
+func (p *AxualProvider) Resources(ctx context.Context) []func() resource.Resource {
+	return []func() resource.Resource{
+		func() resource.Resource { return NewApplicationResource(*p) },
+		func() resource.Resource { return NewUserResource(*p) },
+		func() resource.Resource { return NewGroupResource(*p) },
+		func() resource.Resource { return NewTopicResource(*p) },
+		func() resource.Resource { return NewTopicConfigResource(*p) },
+		func() resource.Resource { return NewEnvironmentResource(*p) },
+		func() resource.Resource { return NewApplicationPrincipalResource(*p) },
+		func() resource.Resource { return NewSchemaVersionResource(*p) },
+		func() resource.Resource { return NewApplicationAccessGrantResource(*p) },
+		func() resource.Resource { return NewApplicationAccessGrantRejectionResource(*p) },
+		func() resource.Resource { return NewApplicationAccessGrantApprovalResource(*p) },
+		func() resource.Resource { return NewApplicationDeploymentResource(*p) },
```

Comment:
Must these resources (and data sources) be prepended with New? Could we go with just the name of the resource itself?

Reply:
Let's not because:

- `applicationResource` is already used inside the resource
- Currently we exactly copy the examples from official and in examples they use `New` because the variable name without `New` is already used: https://github.com/hashicorp/terraform-provider-scaffolding-framework/blob/main/internal/provider/example_resource.go.